src/fastga_he/models/propulsion/components/connectors/h2_fuel_system/unit_tests/test_h2_fuel_system.py

```diff
@@ -0,0 +1,354 @@
+            "data:propulsion:he_power_train:H2_fuel_system:h2_fuel_system_1:CG:y", units="m"
+        ) == pytest.approx(expected_value, rel=1e-2)
+
+        problem.check_partials(compact_print=True)
+
+
+def test_h2_fuel_pipe_length():
+    # Research independent input value in .xml file
+    ivc = get_indep_var_comp(
+        list_inputs(SizingH2FuelSystemLength(h2_fuel_system_id="h2_fuel_system_1")),
+        __file__,
+        XML_FILE,
+    )
+
+    problem = run_system(SizingH2FuelSystemLength(h2_fuel_system_id="h2_fuel_system_1"), ivc)
+
+    assert problem.get_val(
+        "data:propulsion:he_power_train:H2_fuel_system:h2_fuel_system_1:dimension:length", units="m"
+    ) == pytest.approx(4.9, rel=1e-2)
+
+    problem.check_partials(compact_print=True)
+
+
+def test_h2_fuel_inner_diameter():
+    # Research independent input value in .xml file
+    ivc = om.IndepVarComp()
+    ivc.add_output(
+        "data:propulsion:he_power_train:H2_fuel_system:h2_fuel_system_1:pipe_pressure" "",
+        val=70.0,
+        units="MPa",
+    )
+    ivc.add_output(
+        "data:propulsion:he_power_train:H2_fuel_system:h2_fuel_system_1:dimension" ":pipe_diameter",
+        val=0.04,
+        units="m",
+    )
+    ivc.add_output(
+        "data:propulsion:he_power_train:H2_fuel_system:h2_fuel_system_1:material" ":yield_strength",
+        val=240,
+        units="MPa",
+    )
+
+    problem = run_system(SizingH2FuelSystemInnerDiameter(h2_fuel_system_id="h2_fuel_system_1"), ivc)
+
+    assert problem.get_val(
+        "data:propulsion:he_power_train:H2_fuel_system:h2_fuel_system_1:dimension:inner_diameter",
+        units="m",
+    ) == pytest.approx(0.0349, rel=1e-2)
+
+    problem.check_partials(compact_print=True)
+
+
+def test_h2_fuel_inner_diameter_clipped():
+    # Research independent input value in .xml file
+    ivc = om.IndepVarComp()
+    ivc.add_output(
+        "data:propulsion:he_power_train:H2_fuel_system:h2_fuel_system_1:pipe_pressure" "",
+        val=1013250.0,
+        units="Pa",
+    )
+    ivc.add_output(
+        "data:propulsion:he_power_train:H2_fuel_system:h2_fuel_system_1:dimension" ":pipe_diameter",
+        val=0.04,
+        units="m",
+    )
+    ivc.add_output(
+        "data:propulsion:he_power_train:H2_fuel_system:h2_fuel_system_1:material" ":yield_strength",
+        val=240,
+        units="MPa",
+    )
+
+    problem = run_system(SizingH2FuelSystemInnerDiameter(h2_fuel_system_id="h2_fuel_system_1"), ivc)
+
+    assert problem.get_val(
+        "data:propulsion:he_power_train:H2_fuel_system:h2_fuel_system_1:dimension:inner_diameter",
+        units="m",
+    ) == pytest.approx(0.037, rel=1e-2)
+
+    problem.check_partials(compact_print=True)
+
+
+def test_h2_fuel_relative_roughness():
+    # Research independent input value in .xml file
+    ivc = get_indep_var_comp(
+        list_inputs(SizingH2FuelSystemRelativeRoughness(h2_fuel_system_id="h2_fuel_system_1")),
+        __file__,
+        XML_FILE,
+    )
```

Comment:
We've discussed that but for very simple component like that, I'd advise against using the list_inputs func